tensorflow/g3doc/hardware/adding_support/index.md

```diff
@@ -0,0 +1,121 @@
+# Adding Support to New Hardware in TensorFlow
+
+New compute hardware is be release frequently and 
+it is of great interest to do machine learning on 
+this hardware. Much of this hardware is designed 
+with machine learning in mind and had features 
+beyond the normal processor. Taking advantage of
+these novel features is important to make the 
+processor be used to its fullest. For example, the
+[Intel Xeon Phi](http://www.intel.com/content/www/us/en/processors/xeon/xeon-phi-detail.html),
+[KnuPath Hermosa](https://www.knupath.com/products/hermosa-processors/),
+[IBM TrueNorth](http://researchweb.watson.ibm.com/articles/brain-chip.shtml), etc.
+Custom support for each of these different processors
+is needed to make TensorFlow capable of using them.
+
+This is a walk through of how to add support for 
+new hardware to TensorFlow. It is assumed that you
+can link C/C++ code that uses your particular hardware
+into TensorFlow. For simplicity, I will link
+code that implement standard BLAS calls.
+
+## TensorFlow Class Structure
```

Comment:
The exact class structure is probably subject to change in general, so I think it would be better to just point out the location of the Interface (DeviceBase and Device).